scrape/target.go

```diff
@@ -422,41 +399,54 @@ func populateLabels(lset labels.Labels, cfg *config.ScrapeConfig) (res, orig lab
 		lb.Set(model.InstanceLabel, addr)
 	}
 
-	res = lb.Labels()
+	res := lb.Labels()
 	for _, l := range res {
 		// Check label values are valid, drop the target if not.
 		if !model.LabelValue(l.Value).IsValid() {
-			return nil, nil, errors.Errorf("invalid label value for %q: %q", l.Name, l.Value)
+			return nil, errors.Errorf("invalid label value for %q: %q", l.Name, l.Value)
 		}
 	}
-	return res, preRelabelLabels, nil
+	return res, nil
 }
 
-// targetsFromGroup builds targets based on the given TargetGroup and config.
-func targetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig) ([]*Target, error) {
-	targets := make([]*Target, 0, len(tg.Targets))
+// stackLabels stacks labels from different sources for a particular target.
```

Comment:
Stacking is not a term we use, the previous names were fine. There's no stacks here, just a key-value set.